src/utils/callCRMAPI.js

```diff
@@ -88,11 +117,29 @@ export async function addNewSetting() {
   const result = await window.ZOHO.CRM.API.insertRecord({
     Entity: "advancedroundrobin__Round_Robin_Settings",
     APIData: {
-      Name: 'New Setting',
-      advancedroundrobin__Module: 'Leads'
+      Name: "New Setting",
+      advancedroundrobin__Module: "Leads",
     },
-    Trigger: ["workflow"]
+    Trigger: ["workflow"],
   });
 
   return result.data[0];
 }
+
+let roundRobinAvailabilityCache = [];
+
+export async function loadRoundRobinAvailability() {
+  if (!roundRobinAvailabilityCache.length) {
+    await window.ZOHO.embeddedApp.init();
+    const roundRobinAvailabilityResponse = await window.ZOHO.CRM.API.getAllRecords(
```

Comment:
we should use searchRecords here to find the availability record for the setting we're editing

Reply:
@jeznag  I have changed the code to add availability record in the settings itself when we load the settings for first time